flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/FlinkLogicalRelFactories.scala

```diff
@@ -174,6 +174,7 @@ object FlinkLogicalRelFactories {
     def createCorrelate(
         left: RelNode,
         right: RelNode,
+        hints: util.List[RelHint],
```

Comment:
This change came from https://issues.apache.org/jira/browse/CALCITE-4967

Reply:
Is there a Flink ticket to implement hints and verify that they work?

Reply:
It looks it was implemented in Flink in a different way within https://issues.apache.org/jira/browse/FLINK-29280.
And it was one of the reason of having Calcite's `RelDecorrelator` in Flink's repo.
May be it would make sense to create a separate issue to move to Calcite's hints implementation... (probably after update to 1.31.0 where the major part of hint support was added in Calcite)
Howbeit there are still other issues requiring Calcite's `RelDecorrelator` in Flink's repo.